opengrok-web/src/main/java/org/opengrok/web/GetFile.java

```diff
@@ -29,24 +29,28 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.util.Objects;
 
 import jakarta.servlet.http.HttpServlet;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
 import org.opengrok.indexer.history.HistoryGuru;
 import org.opengrok.indexer.web.Prefix;
 
+/**
+ * Used by the webapp to serve the contents of files on /raw and /download.
+ */
 public class GetFile extends HttpServlet {
 
-    public static final long serialVersionUID = -1;
+    private static final long serialVersionUID = -1;
 
     @Override
     protected void service(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
         PageConfig cfg = PageConfig.get(request);
         cfg.checkSourceRootExistence();
 
         String redir = cfg.canProcess();
-        if (redir == null || redir.length() > 0) {
```

Comment:
Would it be more efficient to use an early return for both conditions?
if(redir==null){
 response.setnotfound):
 return;
}
if(!redir.isEmpty()){
 response.sendRedirect(redir);
 return;
}

Reply:
looks much more readable, thanks !